.changeset/cold-schools-yell.md

```diff
@@ -0,0 +1,34 @@
+---
+'astro': minor
+---
+
+Add integrations dynamically
+
+Astro integrations can now dynamically add additional integrations during set-up. This makes it possible for integration authors to bundle integrations more intelligently for their users.
+
+For example, an integration can decide to add Astro’s official sitemap integration if it sees a user hasn’t already configured it:
```

Comment:
```suggestion
In the following example, an Astro integration's own configuration includes a check for Astro’s official sitemap integration. If not found, the integration will add `@astrojs/sitemap` to the existing project, passing any desired configuration options:
```
Was trying to see whether we could use our newer suggestion format here...

Reply:
Tried to rework a bit to de-clunkify it?
```suggestion
In the following example, a custom integration checks if `@astrojs/sitemap` is already configured. If not, the integration adds Astro’s sitemap integration, passing any desired configuration options:
```

Reply:
"checks whether" and yes, I think it's good!